### ISSUE TITLE
Improve Location search (Message "No results found", send Form via Enter)

### ISSUE THREAD

**Issue by tordans** [openstreetmap/iD#6932]
<img width="367" alt="Bildschirmfoto 2019-10-10 um 07 01 00" src="https://user-images.githubusercontent.com/111561/66540336-c43d8d00-eb2b-11e9-8646-d9bbc50ccd1d.png">

1. I am at http://preview.ideditor.com/master/#background=Berlin-2019&disable_features=boundaries&locale=en&map=20.48/52.47762/13.44972
2. I type "Schlesische Straße, Berlin"
3. UI shows "No results found"
4. I press "Enter" => Nothing happens
5. I click the "Search worldwide" Button => Results found

I have different suggestions for this …

### Allow "Enter" to execute the "Search worldwide" button

To me this feels like the natural way of continuing in the form.

### Improve the error Message "No results found"

The way I interpret this message, it actually means "So result found in the data that we downloaded and have in cache". Is that correct?

### Consider a different behavior of the UI / more general

I wonder, if the UI should not execute a general search every time.
If the "search in local cache" is really an important usecase (what would be a good test case for this?), the UI could still do this first and maybe report results faster and in a separate group.

But I don't see a reason why …
- the UI should not search "worldwide" any time I search
- execute and show the results of a "worldwide" search as soon as the local search did not have results (aka "auto-click the 'search worldwide' button).

**Comment by eehpcm** [comment]
@tordans 

> But I don't see a reason why …

>     the UI should not search "worldwide" any time I search

>    execute and show the results of a "worldwide" search as soon as the local search did not have results (aka "auto-click the 'search worldwide' button).

Imagine that you are searching for "Foo" which is in the visible area of the map, but the label isn't showing for some reason (which is why you needed to search for it).  But you made a typo, and entered "Fii" which isn't in the visible area of the map but is somewhere else on the planet.  With your suggestion iD "helpfully" takes you to a very distant place on the map and you may have no easy way back to where you were, other than by repeated panning/zooming to home in on it.

Full auto would be bad.  But pressing enter again would work for me.  Or even shift-enter to make certain it doesn't happen by mistake.  I like keyboard shortcuts that eliminate mouse movement.

**Comment by quincylvania** [comment]
@tordans I agree it'd be nice to always show worldwide results alongside local ones. Though I'm not sure what the constraints are for the lookup server and if it can suddenly take 10x the requests from iD.

> With your suggestion iD "helpfully" takes you to a very distant place on the map and you may have no easy way back to where you were, other than by repeated panning/zooming to home in on it.

@eehpcm I don't think anyone's suggesting iD should move the map as you type, just run the query and show the results in the list.

**Comment by eehpcm** [comment]
@quincylvania 

>  I don't think anyone's suggesting iD should move the map as you type, just run the query and show the results in the list.

I didn't think he was suggesting moving the map as you type.  He was talking about automatically going straight to a worldwide search _after pressing enter_.  Which is a bad idea, because many of us make typos without realizing then press enter.  Or make a typo and press enter automatically before realizing that we made a typo.  So a level of confirmation is desirable, in case our typo has a hit elsewhere in the world, just going straight to a worldwide search on the first enter is a bad idea.

**Comment by tordans** [comment]
> Though I'm not sure what the constraints are for the lookup server and if it can suddenly take 10x the requests from iD.

Yes, it looks like this is an issue. The way I interpret the usage policy at  https://operations.osmfoundation.org/policies/nominatim/ is, that a website/app should not send more that 1 request per second (even though the rate limit actually is per user/ip). In any case, IMO it's not worth the risk of running into issues on this front.

(Example search string `https://nominatim.openstreetmap.org/search/Schlesische%20Stra%C3%9Fe?limit=10&format=json`)

**Comment by quincylvania** [comment]
> a website/app should not send more that 1 request per second

iD could still do live lookup but throttle within the limit (assuming it's actually per-user), though this could appear sluggish. But we should probably just check with the folks running the server to see what the current practical limits are. After all, mappers generally aren't searching that often.

**Comment by abhisheknaiidu** [comment]
hi @quincylvania, by adding some snippet code in iD.js, it is somewhat responding the same as defined, but I'm not able to figure it out which function should be called inside the if statement.


                               
                               
                       var feature_search = document.querySelector('.feature-search');
			feature_search.addEventListener("keyup", function(event) {
			event.preventDefault();
			if (event.keyCode === 13) {
					
			var x = document.getElementsByClassName("geocode-item secondary");
                        x.addEventListener('click', () => {            
		
				});
			}
		});
